### PR TITLE
dev: refactor kakarot_rpc_core::test_utils to a kakarot_test_utils crate

### PR DESCRIPTION
**Pull request by bajpai244**
Time spent on this PR:

Resolves: #554 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

The kakarot_rpc_core::test_utils has been refactored to a new trait `kakarot_test_utils`

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/core/src/client/mod.rs [hunk 1]
```diff
@@ -62,7 +62,7 @@ use crate::models::transaction_receipt::StarknetTransactionReceipt as Transactio
 use crate::models::ConversionError;
 
 pub struct KakarotClient<P: Provider + Send + Sync> {
-    starknet_provider: Arc<P>,
+    pub starknet_provider: Arc<P>,
```
Comment (greged93) [inline crates/core/src/client/mod.rs:65]: Any reason for the pub?

File: crates/test-utils/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+[package]
```
Comment (greged93) [inline crates/test-utils/Cargo.toml:1]: Seems we have two manifeste in test-utils, what's the reason for this? Also the dump-katana bin is define twice

File: crates/core/src/client/mod.rs (hunk 1 above)
Comment (bajpai244) [reply crates/core/src/client/mod.rs:65]: Ah, this was earlier, when I was dealing with the circular dependency issue, this should not be here. changing

File: crates/test-utils/Cargo.toml (hunk 2 above)
Comment (bajpai244) [reply crates/test-utils/Cargo.toml:1]: This was copied by mistake, thanks for catching this, it has been removed via [this](https://github.com/kkrt-labs/kakarot-rpc/pull/553/commits/acd1869260b4d43af4642ce5624f8a0bdbb3f592) commit.

File: crates/core/src/client/mod.rs (hunk 1 above)
Comment (bajpai244) [reply crates/core/src/client/mod.rs]: resolved [here](https://github.com/kkrt-labs/kakarot-rpc/pull/553/commits/acd1869260b4d43af4642ce5624f8a0bdbb3f592).

File: crates/test-utils/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+bytes = "1"
+dotenv = { workspace = true }
+ethers = { workspace = true }
+ethers-solc = { workspace = true }
+foundry-config = { git = "https://github.com/foundry-rs/foundry", branch = "master" }
+rstest = { workspace = true }
+git2 = { workspace = true, optional = true }
+
+futures = "0.3.26"
+serde = { version = "1.0" }
```
Comment (greged93) [inline crates/test-utils/Cargo.toml:36]: Can't you use serde and serde json from the workspace ?

**Review by greged93 — COMMENTED** [review]
Just one comment then looks good

File: crates/test-utils/Cargo.toml (hunk 3 above)
Comment (bajpai244) [reply crates/test-utils/Cargo.toml]: So, I dependencies are copied from the old toml file, but I changed them to workspace, and the tests still seem to be passing, so I have updated them via [this](https://github.com/kkrt-labs/kakarot-rpc/pull/553/commits/cafa69242c75540fa7be96a2c1b455434645d6ad) commit.

File: crates/test-utils/Cargo.toml [hunk 4]
```diff
@@ -33,8 +33,8 @@ rstest = { workspace = true }
 git2 = { workspace = true, optional = true }
 
 futures = "0.3.26"
-serde = { version = "1.0" }
-serde_json = { version = "1.0", features = ["preserve_order"] }
+serde = { workspace = true}
```
Comment (greged93) [inline crates/test-utils/Cargo.toml:36]: Just fix the formatting issue, then lgtm 
workspace = true }

Comment (bajpai244) [reply crates/test-utils/Cargo.toml:36]: Ah, interesting, cargo fmt should also do that! Don't know why it doesn't.

Comment (bajpai244) [reply crates/test-utils/Cargo.toml]: Done [here](https://github.com/kkrt-labs/kakarot-rpc/pull/553/commits/773ff312844dc8d7be2e7b7f986d0a8a6091d027).

**Review by greged93 — APPROVED** [review]
Lgtm!

**Comment by greged93** [comment]
Thanks for the refactor! Now we start cleaning these utils :)

**Review by ClementWalter — APPROVED** [review]

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**